packages/office-ui-fabric-react/src/common/shallowUntilTarget.ts

```diff
@@ -0,0 +1,64 @@
+import { shallow, ShallowWrapper } from 'enzyme';
+
+/**
+ * Duplicated enzyme's ShallowRendererProps
+ *
+ * @internal
+ */
+export interface IShallowRendererProps {
+  lifecycleExperimental?: boolean;
+  disableLifecycleMethods?: boolean;
+}
+
+/**
+ * ShallowUntilTarget Interface
+ *
+ * @internal
+ */
+export interface IShallowUntilTarget {
+  maxTries: number;
+  shallowOptions: IShallowRendererProps;
+}
+
+/**
+ * An extention of enzyme's shallow function which will fail to work
+ * with decorated components and/or components using the styled() function.
+ * This function allows you to pass a 'target' component (e.g. ComponentBase)
+ * and keep running shallow on each child component till a match is found.
+ *
+ * @public
+ */
+export function shallowUntilTarget<P, S>(
+  componentInstance: React.ReactElement<P>,
+  TargetComponent: string,
+  options: IShallowUntilTarget = {
+    maxTries: 10,
+    shallowOptions: {}
+  }
+): ShallowWrapper {
+  const { maxTries, shallowOptions } = options;
+
+  let root = shallow<P, S>(componentInstance, shallowOptions);
+
+  if (typeof root.type() === 'string') {
+    // If type() is a string then it's a DOM Node.
+    // If it were wrapped, it would be a React component.
+    throw new Error('Cannot unwrap this component because it is not wrapped');
+  }
+
+  for (let tries = 1; tries <= maxTries; tries++) {
+    // Check for target as a string to avoid conflicts
+    // with decoratored components name
+    if (root.type().toString().includes(TargetComponent)) {
+      // Now that we found the target component, render it.
+      return root.first().shallow(shallowOptions);
+    }
+    // Unwrap the next component in the hierarchy.
+    root = root.first().shallow(shallowOptions);
```

Comment:
While I don't think that any decorators we currently have render multiple siblings, it is possible that this will miss a component that isn't the first child.